frontend/src/modules/WellCompletions/settings.tsx

```diff
@@ -116,7 +147,14 @@ export const settings = ({ moduleContext, workbenchSession, workbenchServices }:
             }
             createAndSetPlotData(allTimeSteps, timeStepIndex, selectedTimeStepOptions.timeAggregationType);
         },
-        [wellCompletionsQuery.data, selectedTimeStepOptions]
+        [
+            wellCompletionsQuery.data,
+            selectedTimeStepOptions,
+            availableTimeSteps,
```

Comment:
Wouldn't this re-trigger the useEffect when line 109 and 119 setAvailableTimeSteps?